extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -18,16 +18,50 @@ function realTimeConfigManager(element, win, ampDoc, customMacros) {
   }
   const promiseArray = [];
   const urlToCalloutMap = {};
-  maybeInflatePublisherUrls(rtcConfig, urlToCalloutMap, customMacros, ampDoc);
-  maybeInflateVendorUrls(rtcConfig, urlToCalloutMap, ampDoc);
+  const seenUrls = [];
   const rtcStartTime = Date.now();
-  for (let url in urlToCalloutMap) {
-    promiseArray.push(sendRtcCallout_(
-        url, rtcStartTime, win, rtcConfig['timeoutMillis'], urlToCalloutMap[url]));
-  }
+  (rtcConfig['urls'] || []).forEach(url => {
+    inflateAndSendRtc_(element, url, seenUrls, promiseArray,
+                       rtcStartTime, customMacros, win, ampDoc,
+                       rtcConfig['timeoutMillis']);
+  });
+  Object.keys(rtcConfig['vendors'] || []).forEach(vendor => {
+    const url = RTC_VENDORS[vendor.toLowerCase()];
+    if (!url) {
+      user().warn(TAG, `unknown vendor ${vendor}`);
+      return;
+    }
+    const macros = rtcConfig['vendors'][vendor];
+    inflateAndSendRtc_(element, url, seenUrls, promiseArray, rtcStartTime,
+                       macros, win, ampDoc, rtcConfig['timeoutMillis'],
+                       vendor);
+  });
   return Promise.all(promiseArray);
 }
 
+function inflateAndSendRtc_(element, url, seenUrls, promiseArray, rtcStartTime,
+                            macros, win, ampDoc, timeoutMillis, opt_vendor) {
+  if(promiseArray.length == MAX_RTC_CALLOUTS) {
+    dev().warn(TAG, `${MAX_RTC_CALLOUTS} RTC Callout URLS exceeded, ` +
+               ` dropping ${url}`);
+    return;
+  }
+  const urlReplacements = Services.urlReplacementsForDoc(ampDoc);
+  // TODO: change to use whitelist.
+  url = urlReplacements.expandSync(url, macros);
+  const callout = opt_vendor || url;
+  try {
+    user().assert(isSecureUrl(url),
```

Comment:
We do this enough that we should consider creating a helper function:

// @param {!Array!<<pred: function():boolean, log: string>>} checks
function assertAndLog(checks) {
    for (let i = 0; i < checks.length; i++) {
        const check = checks[i];
        const result = ;
        if (!check.predicate()) {
           user().warn(TAG, check.message);
           return false;
        }
     }
     return true;
}
if (!assertAndLog([
     {pred: () => isSecure(url), log: `Dropping RTC URL: ${url}, not secure`}, 
     {pred: () => !seenUrls.includes(url), log: `Dropping duplicate calls to RTC URL: ${url}`}])) {
   return;
}